mule-extensions-api/src/main/java/org/mule/runtime/extension/api/annotation/ConfigReference.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.annotation;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to be used in a {@link String} type {@link Parameter} field or parameter that is an
+ * {@link ConfigReference}.
```

Comment:
ConfigReference is not an actual class... The overall grammar of this jdoc is not super clear. How about something more simple such as `Annotation to be used on parameters which value should reference a global config element`